### ISSUE TITLE
Add in component Sass/JavaScript building

### ISSUE THREAD

**Issue by rowanmanning** [Financial-Times/origami-registry-ui#12]
This service, unlike the others, needs to install and build components locally and compile into CSS/JavaScript bundles. We currently use the Build Service for this elsewhere, but the registry has a lot more custom styling and JavaScript.

There are a couple of ways we could do this, I think both should use OBT. If you can think of more then that's fine, these are just the top-of-my-head options:

1. You manually compile the CSS/JS (or have a watch script). Compiled resources are committed to the repo just like the source files. **Pros:** simpler to implement. **Cons:** easy to forget to do, generated files committed to repo.

2. The service compiles the CSS/JS when it starts up. Compiled resources are not committed. **Pros:** No generated files in the repo, resources are always up to date. **Cons:** Slower startup times, Possible for CSS change to take the service down.

In terms of where stuff lives:
  - We'll need OBT as a dependency (either dev or prod, depending on which route we take)
  - We'll need to install Bower dependencies as well as Node.js modules
  - If option 1, then a few `Makefile` updates might be enough
  - If option 2, then `lib/service.js` will probably need some edits, and possibly we'd want the CSS/JS compilation to be in a separate `lib` file so that we can easily move into `origami-service` later.

**Comment by gvonkoss** [comment]
3. We build locally and ignore the files on push, and have the app bundle our assets in production (if something breaks at that point, we don't get a broken release), so a combination of 1 + 2.

**Comment by JakeChampion** [comment]
Note: Next uses [haikro](https://github.com/matthew-andrews/haikro) which does not work with the Heroku-GitHub integration that we use in all our other services.

**Comment by rowanmanning** [comment]
OK based on a couple of face-to-face discussions, I _think_ we decided the following. @gvonkoss, let me know if I've got any of this wrong.

1. There will be a new make task. When you run this make task, Sass and JavaScript will be bundled together and saved somewhere like the `/public` folder. These bundled files will be ignored by git.
2. This make task will be run during the `release` phase on Heroku (or npm postinstall, not sure) so that the bundled files are present on the dyno. This has the benefit of failing a deploy if the bundles fail to compile – no broken CSS/JS makes it to production
3. There will be some way of watching for file changes locally and running this task when Sass or JavaScript files change. Possibly we could integrate this with the `make start-dev` task, so that you only have to run one thing in local development
